SQL-compiler/src/main/java/org/dbsp/sqllogictest/executors/DBSPExecutor.java

```diff
@@ -140,20 +143,48 @@ DBSPFunction createInputFunction(TableValue[] tables) throws IOException {
 
         // If the data is large write it to a set of CSV files and read it at runtime.
         if (totalSize > 10) {
-            for (int i = 0; i < tables.length; i++) {
-                String fileName = (rustDirectory + tables[i].tableName) + ".csv";
-                Solutions.toCsv(fileName, tables[i].contents);
-                fields[i] = new DBSPApplyExpression("read_csv",
-                        tables[i].contents.getNonVoidType(),
-                        new DBSPStrLiteral(fileName));
+            if (connectionString.equals("csv")) {
+                for (int i = 0; i < tables.length; i++) {
+                    String fileName = (rustDirectory + tables[i].tableName) + ".csv";
+                    Solutions.toCsv(fileName, tables[i].contents);
+                    fields[i] = new DBSPApplyExpression("read_csv",
+                            tables[i].contents.getNonVoidType(),
+                            new DBSPStrLiteral(fileName));
+                }
+            } else {
+                // read from DB
+                for (int i = 0; i < tables.length; i++) {
+                    fields[i] = generateReadDbCall(tables[i]);
+                }
             }
         }
-
         DBSPRawTupleExpression result = new DBSPRawTupleExpression(fields);
         return new DBSPFunction("input", new ArrayList<>(),
                 result.getType(), result);
     }
 
+    private DBSPExpression generateReadDbCall(TableValue tableValue) {
+        // Generates a read_table(<conn>, <table_name>, <mapper from |AnyRow| -> Tuple type>) invocation
+        DBSPTypeUser sqliteRowType = new DBSPTypeUser(null, "AnyRow", false);
+        DBSPVariablePath rowVariable = new DBSPVariablePath("row", sqliteRowType);
+        DBSPTypeTuple tupleType = (DBSPTypeTuple) tableValue.contents.zsetType.elementType;
```

Comment:
use the .to method instead of a cast, it's nicer.

Reply:
Done.